models/proof.rb

```diff
@@ -0,0 +1,34 @@
+class Proof < Sequel::Model
+  def validate
+    super
+
+    errors.add(:mass_g, mass_outweighed_error_msg) if mass_outweighs_offset?
+  end
+
+  def after_create
+    capture_payments
+    offset.retire!(mass_g)
+
+    super
+  end
+
+  private
+
+  def offset
+    @offset ||= Offset[offset_id]
+  end
+
+  def mass_outweighs_offset?
+    return true if mass_g > offset[:mass_g]
+    false
```

Comment:
[nit] Doesn't seem like there is a need for the `if` or guard clause
```suggestion
    mass_g > offset[:mass_g]
```